osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -204,27 +207,41 @@ protected override void LoadComplete()
         /// <param name="lifetimeEntry">The <see cref="HitObjectLifetimeEntry"/> controlling the lifetime of <paramref name="hitObject"/>.</param>
         public void Apply([NotNull] HitObject hitObject, [CanBeNull] HitObjectLifetimeEntry lifetimeEntry)
         {
-            free();
-
-            HitObject = hitObject ?? throw new InvalidOperationException($"Cannot apply a null {nameof(HitObject)}.");
-
-            this.lifetimeEntry = lifetimeEntry;
+            if (hitObject == null)
+                throw new ArgumentNullException($"Cannot apply a null {nameof(HitObject)}.");
 
             if (lifetimeEntry != null)
             {
-                // Transfer lifetime from the entry.
-                LifetimeStart = lifetimeEntry.LifetimeStart;
-                LifetimeEnd = lifetimeEntry.LifetimeEnd;
+                if (lifetimeEntry.HitObject != hitObject)
+                    throw new InvalidOperationException($"{nameof(HitObjectLifetimeEntry)} has different {nameof(HitObject)} from the specified one.");
```

Comment:
Not super sure about this intermediate state of things where runtime exceptions are getting thrown on bad input... Sure, obsoleting this method right now in this PR would hurt, but it'll hurt and be a huge mess either way. Would probably rather see a new overload that only accepts the entry in here too? Or, at least it can be added, without the obsoletion of the old one for now...

Reply:
I would like to wait to deprecate things until a more complete picture of the situation is drawn. I don't want to make breaking changes for just later realizing a different change was better.

Reply:
That being said, I added public `Apply(HitObjectLifetimeEntry)` and `Apply(HitObject)` as I agreed the state of having no access to the "correct" method is bad.

Reply:
Honestly, that non-committal attitude towards deprecating the existing overload is not giving me confidence in approving this change. I personally need to see a cold hard diff with the deprecated variant removed, and as a result see how the existing usages will behave with the new code paths going forward. Otherwise, unfortunately I am nowhere near smart enough to be able to tell just from my head alone if the new overloads are a good direction or not. Maybe the others are, I don't know.

Reply:
The change required for the deprecation of this overload is fairly trivial, and I just pushed that change.
My first comment above was probably misleading, but it was a comment about my general objective of not making unnecessary breaking changes in PRs, not specific to this change.